### ISSUE TITLE
Cannot parse yaml file

### ISSUE THREAD

**Issue by RafaelGCPP** [goccy/go-yaml#118]
My config file cannot be parser. It complains on a unexpected key at 16,1.  Removing the empty line before the "masters:" key, makes Unmarshall parse it.

[bad.yml.txt](https://github.com/goccy/go-yaml/files/4742287/bad.yml.txt)

The yamlv.2 package parses it correctly, though.

**Comment by goccy** [comment]
I could not reproduce this problem ( my library's version is `v1.7.1` ) .

Could you tell me your library's version ( also Go version ) ？

Also, maybe your YAML file is invalid ( `*global-config` recursively call own )

**Comment by RafaelGCPP** [comment]
I'll check here, but somehow masters: got indented. In the file it is not a key inside global-config..

**Comment by goccy** [comment]
I got the following .

```yaml
dmrid: &dmrid 123456789

config: &global-config
  callsign: M1ABC
  rxfreqmhz: 431.000000
  txfreqmhz: 431.000000
  txpower: 1
  colorcode: 1
  latitude: -22.92657
  longitude: -43.22793
  height: 0
  location: "Rio de Janeiro, GG87"
  description: Brazil
  url: "www.qrz.com/m1abc"

  masters:
  - netid: BM
    name: Brandmeister Brazil
    address: 191.252.185.53:62031
    password: mypassword
    dmrid: *dmrid
    config: *global-config

  - netid: TGIF
    name: TGIF Network
    address: tgif.network:62031
    password: mypassword
    dmrid: *dmrid  
    config: *global-config

  - netid: DMR+
    name: DMR+ IPSC2-BRAZIL
    address: 201.62.48.62:55555
    password: mypassword
    dmrid: *dmrid  
    config: *global-config
```

Please directly paste your YAML to here .

**Comment by RafaelGCPP** [comment]
I'm using go 1.14 and version 1.17.1 of your library. It fails complaining of a 'unexpected key' at line 16 col 1. If I remove the blank line 15, it parses ok.  This is the version of the configuration that fails in my code:

```yaml
dmrid: &dmrid 123456789

config: &global-config
  callsign: M1ABC
  rxfreqmhz: 431.000000
  txfreqmhz: 431.000000
  txpower: 1
  colorcode: 1
  latitude: -22.92657
  longitude: -43.22793
  height: 0
  location: "Rio de Janeiro, GG87"
  description: Brazil
  url: "www.qrz.com/m1abc"

masters:
  - netid: BM
    name: Brandmeister Brazil
    address: 191.252.185.53:62031
    password: mypassword
    dmrid: *dmrid
    config: *global-config

  - netid: TGIF
    name: TGIF Network
    address: tgif.network:62031
    password: mypassword
    dmrid: *dmrid  
    config: *global-config

  - netid: DMR+
    name: DMR+ IPSC2-BRAZIL
    address: 201.62.48.62:55555
    password: mypassword
    dmrid: *dmrid  
    config: *global-config```

The other one that got upload was really and obviously wrong. I still don't get how it got that way...

**Comment by goccy** [comment]
Hmmm.. I could parse your pasted YAML ( exists blank line ) my environment .
Maybe.... You have executed this program on Windows ?
I guess affected by `CR/LF`

**Comment by goccy** [comment]
Work this on go playground

https://play.golang.org/p/9PBIk2h0E3H

**Comment by RafaelGCPP** [comment]
Yes, I am doing development on Windows. This code will be run in multiple platforms...

**Comment by goccy** [comment]
I fixed this issue with #127 and released as v1.7.5 !
Thanks !!

**Comment by RafaelGCPP** [comment]
You are very much welcome!
